ledger/ledgercore/accountresource.go

```diff
@@ -22,44 +22,37 @@ import (
 
 // AccountResource used to retrieve a generic resource information from the data tier
 type AccountResource struct {
-	CreatableIndex basics.CreatableIndex
-	CreatableType  basics.CreatableType
-
 	AssetParams   *basics.AssetParams
 	AssetHolding  *basics.AssetHolding
 	AppLocalState *basics.AppLocalState
 	AppParams     *basics.AppParams
 }
 
-// AssignAccountData assigned the Asset/App params/holdings contained in the AccountResource
-// to the given basics.AccountData, creating maps if necessary.
-func (r *AccountResource) AssignAccountData(ad *basics.AccountData) {
-	switch r.CreatableType {
-	case basics.AssetCreatable:
-		if r.AssetParams != nil {
-			if ad.AssetParams == nil {
-				ad.AssetParams = make(map[basics.AssetIndex]basics.AssetParams)
-			}
-			ad.AssetParams[basics.AssetIndex(r.CreatableIndex)] = *r.AssetParams
+// AssignAccountResourceToAccountData assignes the Asset/App params/holdings contained
+// in the AccountResource to the given basics.AccountData, creating maps if necessary.
+func AssignAccountResourceToAccountData(cindex basics.CreatableIndex, resource AccountResource, ad *basics.AccountData) {
```

Comment:
maybe we move AccountResource to `basics` package and and add 
```
func (ad * AccountData) AssignResource(cidx CreatableIndex, resource AccountResource)
```
there?

Reply:
@tsachiherman, @cce are you ok with it? I don't know what your plans for `basics.AccountData` are.

Reply:
We can do it in another PR since it can be large.

Reply:
I don't think that we can move this to `basics` due to the `AccountResource`, but I'm wondering about the choice of having this as independent method instead of a member function:
```golang
func (r *AccountResource) AssignAccountData(ad *basics.AccountData, cidx basics.CreatableIndex)
```

Reply:
I had assumed some of these types will need to move into basics, AccountResource seems pretty harmless to move in there..

Reply:
I'm not sure if `AccountResource` is a good candidate for an object that lives outside the ledger package. The fact that we've been able to keep it in ledgercore is a good indication that it probably *shouldn't* be exposed. The *content* of that structure might need to be exported, through.